### PR TITLE
Example of what automated PR doc checking could look like

### PR DESCRIPTION
**Pull request by JohnHillegass**
For #311

This was a joint effort with @JonZeolla. Looking forward to feedback and further discussion on how we can implement it.  

This PR adds a Makefile with document check commands which could be run locally or remotely on Github actions. It adds an editorconfig for editor level checks, linting (markdownlint) and link checking (markdown-link-check) for markdown files and a spellcheck (cspell).

To run these commands locally, you will need `docker-compose` installed and can then just run the commands listed in the Makefile. For example, to run the lint checks we would use `make lint` as a terminal command.

For implementation, I suggest setting branch protection rules as shown below so that they do not block PR merges but will call out any issues. 

image

### CODE REVIEW

**Comment by lumjjb** [comment]
This is awesome @JohnHillegass!! Thanks so much, is it easy to create a Makefile command or something that can be used to test locally?

**Comment by JohnHillegass** [comment]
Oh yes, a make command works for me. I was not sure what the appetite would be for running locally, needing node dependencies and what not. I think we can just communicate install steps though.

**Comment by lumjjb** [comment]
Yea - I think at least initially it would be super helpful since there's
likely going to be a LOT of cleanup :). Once we have something we can
probably create a containerized version, that would work nicely as well!

File: cspell.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        "RBAC",
+        "Roadmap",
+        "ultrasaurus",
+        "Jeyappragash",
+        "pragashj",
+        "dshaw",
+        "lumjjb",
+        "Cappos",
+        "ericavonb",
+        "HIPPA",
```
Comment (JonZeolla) [inline cspell.json:23]:
```suggestion
        "HIPAA",
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 We aspire to create a welcoming environment for collaboration on this project
 and ask that all contributors do the same. For more details, see our
 [code of conduct](CODE-OF-CONDUCT.md).
+   blehblel *
```
Comment (JonZeolla) [inline CONTRIBUTING.md:6]:
```suggestion
```

**Comment by JonZeolla** [comment]
> Oh yes, a make command works for me. I was not sure what the appetite would be for running locally, needing node dependencies and what not. I think we can just communicate install steps though.

We should be able to have the make rule run a container and volume mount `$(pwd)` into it to reduce host dependencies and enforce version pinning.

**Review by JonZeolla — CHANGES_REQUESTED** [review]
Please consider https://github.com/JohnHillegass/sig-security/pull/2

**Comment by JohnHillegass** [comment]
Great points @JonZeolla. Thanks for adding these changes.

**Comment by lumjjb** [comment]
@JohnHillegass @JonZeolla Do ping here if you think its ready for us to try out! And let's take it out of draft and start reviewing so we can get this awesome feature in!

**Comment by JohnHillegass** [comment]
@lumjjb I think it's ready to go now! @JonZeolla's refactor makes it very easy to run both locally and remotely with Make and docker-compose.

**Comment by JohnHillegass** [comment]
@achetal01 thanks for the suggestion today for updating this with a summary. I have added some more details to the first comment in the PR.

**Comment by JohnHillegass** [comment]
superseded by #497

**Pull request closed without merging**